packages/augur-tools/src/flash/scripts.ts

```diff
@@ -1076,44 +1084,102 @@ export function addScripts(flash: FlashSession) {
         flag: true,
       },
       {
-        name: 'repeats',
-        abbr: 'r',
-        description: 'how many times to take. default=1',
+        name: 'userAccount',
+        abbr: 'u',
+        description: 'user account to create the order',
       },
       {
-        name: 'max',
+        name: 'outcome',
+        abbr: 'o',
+        description: 'orderbook outcome to take, default is 2'
+      },
+      {
+        name: 'market',
         abbr: 'm',
-        description: 'how many orders to take at once. default=10',
+        description: 'market to trade, default is a random market',
+      },
+      {
+        name: 'limit',
+        abbr: 'l',
+        description: 'limit of orders to take, 1...N orders can be take, default is keep taking forever',
       },
       {
         name: 'wait',
         abbr: 'w',
         description: 'how many seconds to wait between takes. default=1',
       },
+      {
+        name: 'direction',
+        abbr: 'd',
+        description: 'side of orderbook to take, bid or ask, ask is default',
+      },
+      {
+        name: 'meshEndpoint',
+        abbr: 'e',
+        required: false,
+        description: 'Mesh endpoint to connect',
+      },
     ],
     async call(this: FlashSession, args :FlashArguments) {
       const skipFaucet = args.skipFaucet as boolean;
-      let repeats = Number(args.repeats as string) || 1;
-      const max = Number(args.max as string) || 10;
-      const wait = Number(args.wait as string) || 1;
+      const address = args.userAccount ? String(args.userAccount) : null;
+      let marketId = args.market ? String(args.market) : null;
+      let limit = args.limit ? Number(args.limit) : 86400000; // go for a really long time
+      const direction = args.direction ? String(args.direction) : 'bid'
+      const outcome = args.outcome ? Number(args.outcome) : 2;
+      const wait = Number(String(args.wait)) || 1;
+
+      const endpoint = args.meshEndpoint
+        ? String(args.meshEndpoint)
+        : 'ws://localhost:60557';
+
+      const user: ContractAPI = await this.ensureUser(null, true, true, address, endpoint, true);
 
-      const user = await this.ensureUser(null, true, true, null, 'ws://localhost:60557', true);
       if (!skipFaucet) {
         const funds = new BigNumber(1e18).multipliedBy(1000000);
         await user.faucet(funds);
         await user.approve(funds);
       }
 
       const markets = (await user.getMarkets()).markets;
-      const market = markets[0];
-
-      for (; repeats > 0; repeats--) {
-        let orders = flattenZeroXOrders(await user.getOrders(market.id));
-        if (orders.length > max) {
-          orders = orders.slice(0, max)
+      const market = marketId ? markets.find(m => m.id === marketId) : markets[0];
+
+      console.log('market props')
+      const orderType = direction === 'bid' || direction === 'buy' ? '0' : '1';
+      let loop = true;
+      while (loop) {
```

Comment:
Why not use a for-loop?

Reply:
Could do that. Initially thought it would loop until cancelled then added the large limit value and keep the loop. 
For loop would be more readable. Will change